docs/source/data/data_catalog.md

```diff
@@ -720,125 +131,45 @@ cars:
-
-In a file like `catalog.py`, you can construct a `DataCatalog` object programmatically. In the following, we are using several pre-built data loaders documented in the [API reference documentation](/kedro_datasets).
-
-```python
-from kedro.io import DataCatalog
-from kedro_datasets.pandas import (
-    CSVDataSet,
-    SQLTableDataSet,
-    SQLQueryDataSet,
-    ParquetDataSet,
-)
-
-io = DataCatalog(
-    {
-        "bikes": CSVDataSet(filepath="../data/01_raw/bikes.csv"),
-        "cars": CSVDataSet(filepath="../data/01_raw/cars.csv", load_args=dict(sep=",")),
-        "cars_table": SQLTableDataSet(
-            table_name="cars", credentials=dict(con="sqlite:///kedro.db")
-        ),
-        "scooters_query": SQLQueryDataSet(
-            sql="select * from cars where gear=4",
-            credentials=dict(con="sqlite:///kedro.db"),
-        ),
-        "ranked": ParquetDataSet(filepath="ranked.parquet"),
-    }
-)
-```
-
-When using `SQLTableDataSet` or `SQLQueryDataSet` you must provide a `con` key containing [SQLAlchemy compatible](https://docs.sqlalchemy.org/en/13/core/engines.html#database-urls) database connection string. In the example above we pass it as part of `credentials` argument. Alternative to `credentials` is to put `con` into `load_args` and `save_args` (`SQLTableDataSet` only).
-
-### Load datasets
-
-You can access each dataset by its name.
-
-```python
-cars = io.load("cars")  # data is now loaded as a DataFrame in 'cars'
-gear = cars["gear"].values
-```
-
-#### Behind the scenes
-
-The following steps happened behind the scenes when `load` was called:
-
-- The value `cars` was located in the Data Catalog
-- The corresponding `AbstractDataSet` object was retrieved
-- The `load` method of this dataset was called
-- This `load` method delegated the loading to the underlying pandas `read_csv` function
-
-### View the available data sources
+A dataset offers versioning support if it extends the [`AbstractVersionedDataSet`](/kedro.io.AbstractVersionedDataSet) class to accept a version keyword argument as part of the constructor and adapt the `_save` and `_load` method to use the versioned data path obtained from `_get_save_path` and `_get_load_path` respectively. In Kedro version 0.18.2, the following datasets support versioning:
+
+- `kedro_datasets.api.APIDataSet`
+- `kedro_datasets.holoviews.HoloviewsWriter`
+- `kedro_datasets.json.JSONDataSet`
+- `kedro_datasets.matplotlib.MatplotlibWriter`
+- `kedro_datasets.networkx.NetworkXDataSet`
+- `kedro_datasets.pandas.CSVDataSet`
+- `kedro_datasets.pandas.ExcelDataSet`
+- `kedro_datasets.pandas.FeatherDataSet`
+- `kedro_datasets.pandas.HDFDataSet`
+- `kedro_datasets.pandas.JSONDataSet`
+- `kedro_datasets.pandas.ParquetDataSet`
+- `kedro_datasets.pickle.PickleDataSet`
+- `kedro_datasets.pillow.ImageDataSet`
+- `kedro_datasets.tensorflow.TensorFlowModelDataSet`
+- `kedro_datasets.text.TextDataSet`
+- `kedro_datasets.spark.SparkDataSet`
+- `kedro_datasets.yaml.YAMLDataSet`
```

Comment:
I'm tempted to remove all this and just say either which datasets do _not_ support versioning or leave it to the reader to work it out by looking at the dataset doc. I'm worried about keeping this updated.

Reply:
Yes this list isn't very maintainable.. if the list of datasets that don't support versioning is very short, let's add that. Otherwise, I think it's fine to just explain how you can tell if a dataset support versioning or not.

Reply:
@merelcht Please could you help me to explain how to tell if it supports versioning or not? Is it because it derives from `AbstractVersionedDataSet`? Could you please give me a sentence or a snippet of code for a reader to look for in a dataset .py file so they can check?

Reply:
I'd say something along the lines of: 
"To verify whether a dataset can undergo versioning, you can examine the dataset class's inheritance within the `kedro-datasets` repository. To determine this, check if the dataset class inherits from the `AbstractVersionedDataSet`. For instance, if you encounter a class like `CSVDataSet(AbstractVersionedDataSet[pd.DataFrame, pd.DataFrame])`, this indicates that the dataset is primed for versioning."